src/software/thunderscope/robot_diagnostics/diagnostics_widget.py

```diff
@@ -0,0 +1,165 @@
+        # instead, widgets are added to the layout which is set for a container
+        # the container is set as the current QScrollArea's widget
+        self.container = QFrame(self)
+        self.container.setLayout(diagnostics_widget_vbox_layout)
+        self.setWidget(self.container)
+        self.setWidgetResizable(True)
+
+    def __control_mode_update_handler(self, mode: ControlMode) -> None:
+        """
+        Handler for managing a control mode update
+        :param mode: The new mode that is being used
+        """
+        self.__control_mode = mode
+        self.handheld_device_handler.refresh(self.__control_mode)
+
+    def __handheld_device_connection_status_update_handler(
+        self, status: HandheldDeviceConnectionStatus
+    ):
+        """
+        Handler for propagating
+        :param status: The new mode that is being used
+        """
+        self.__handheld_device_status = status
+
+    def refresh(self):
+        """
+        Refreshes sub-widgets so that they display the most recent status values.
+        If in handheld mode, then also visually updates driver, dribbler and chicker sliders
+        to the values currently being set by the handheld device.
+        """
+
+        # update controller status view with most recent controller status
+        self.handheld_device_status_widget.refresh(self.__handheld_device_status)
+
+        # update diagnostic ui with most recent recent controller status
+        self.diagnostics_control_input_widget.refresh(self.__handheld_device_status)
+
+        # update all widgets based on current control mode
+        self.drive_dribbler_widget.refresh(self.__control_mode)
+        self.chicker_widget.refresh(self.__control_mode)
+
+        if self.__control_mode == ControlMode.HANDHELD:
+            # visually update driver, dribbler, kick power and chip distance sliders,
+            # based on the controller values
+            motor_control = self.handheld_device_handler.motor_control
+            kick_power = self.handheld_device_handler.kick_power_accumulator
+            chip_distance = self.handheld_device_handler.chip_distance_accumulator
+
+            self.drive_dribbler_widget.set_x_velocity_slider(
+                motor_control.direct_velocity_control.velocity.x_component_meters
+            )
+            self.drive_dribbler_widget.set_y_velocity_slider(
+                motor_control.direct_velocity_control.velocity.y_component_meters
+            )
+            self.drive_dribbler_widget.set_angular_velocity_slider(
+                motor_control.direct_velocity_control.angular_velocity.radians_per_second
+            )
+
+            self.drive_dribbler_widget.set_dribbler_velocity_slider(
+                motor_control.dribbler_speed_rpm
+            )
+
+            self.chicker_widget.set_kick_power_slider_value(kick_power)
+            self.chicker_widget.set_chip_distance_slider_value(chip_distance)
```

Comment:
does this mean that the handheld controller settings will be reflected on the drive_dribbler and chicker widgets?